tests/state_test.go

```diff
@@ -31,6 +35,105 @@ type stateCase struct {
 
 var ripemd = types.StringToAddress("0000000000000000000000000000000000000003")
 
+//nolint
+type accountTrie interface {
+	Get(k []byte) ([]byte, bool)
+}
+
+//nolint
+type legacyAccount struct {
+	Nonce    uint64
+	Balance  *big.Int
+	Root     types.Hash
+	CodeHash []byte
+	Trie     accountTrie
+}
+
+//nolint
+func (a *legacyAccount) marshalWith(ar *fastrlp.Arena) *fastrlp.Value {
+	v := ar.NewArray()
+	v.Set(ar.NewUint(a.Nonce))
+	v.Set(ar.NewBigInt(a.Balance))
+	v.Set(ar.NewBytes(a.Root.Bytes()))
+	v.Set(ar.NewBytes(a.CodeHash))
+
+	return v
+}
+
+var stateTestArenaPool fastrlp.ArenaPool //nolint
+
+//nolint
+func hashIt(k []byte) []byte {
+	h := sha3.NewLegacyKeccak256()
+	h.Write(k)
+	return h.Sum(nil)
+}
+
+//nolint
+type coinbaseParams struct {
+	coinbaseAddress types.Address
+	shouldInclude   bool
+}
+
+//nolint
+func pruneStakedAccounts(
+	snapshot state.Snapshot,
+	accountMap map[types.Address]*chain.GenesisAccount,
+	cParms coinbaseParams,
+) []byte {
+	localTrie := itrie.NewTrie()
+	trieTransaction := localTrie.Txn()
+
+	arena := stateTestArenaPool.Get()
+	defer stateTestArenaPool.Put(arena)
+
+	var touchedAccounts []types.Address
+	for accountAddress := range accountMap {
+		touchedAccounts = append(touchedAccounts, accountAddress)
+	}
+
+	if cParms.shouldInclude {
+		touchedAccounts = append(touchedAccounts, cParms.coinbaseAddress)
+
+	}
+
+	for _, accountAddress := range touchedAccounts {
+		key := keccak.Keccak256(nil, accountAddress.Bytes())
+
+		// Grab the account from storage, as it was committed earlier
+		result, ok := snapshot.Get(key)
+		if !ok {
+			return nil
+		}
+
+		var account state.Account
+		if err := account.UnmarshalRlp(result); err != nil {
+			return nil
+		}
+
+		// Create a new legacy account object,
+		// without the staked balance field
+		legacyAcc := &legacyAccount{
+			Nonce:    account.Nonce,
+			Balance:  account.Balance,
+			Root:     account.Root, // Storage root
+			CodeHash: account.CodeHash,
+			Trie:     account.Trie,
+		}
+
+		vv := legacyAcc.marshalWith(arena)
+		data := vv.MarshalTo(nil)
+
+		// Add the legacy account to the local trie
+		trieTransaction.Insert(hashIt(accountAddress.Bytes()), data)
+		arena.Reset()
+	}
+
+	newRoot, _ := trieTransaction.Hash()
+
+	return newRoot
+}
```

Comment:
Maybe remove these changes as well, since `pruneStakedAccounts` isn't used in this PR?

Reply:
Agreed. This code will still be present on the fix branch